apps/05Mastermind.js

```diff
@@ -29,17 +31,51 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
+function generateHint(solution, guess) {
   // your code here
+  var solutionArray = solution.split("");
+  var guessArray = guess.split("");
+  var correctLetterLocations = 0;
+  var correctLetters = 0;
+//could i do this
+//solution is abcd = guess is acde
+for (var i = 0; i < solutionArray.length; i++){
+  if(solutionArray[i] === guessArray[i]){
+  correctLetterLocations++;
+  solutionArray[i] = null;
+  }
+ }
+
+  for (var i= 0; i < solutionArray.length; i++){
+  var targetIndex = solutionArray.indexOf(guessArray[i]);
+  if (targetIndex > -1){
+  correctLetters++;
+  solutionArray[targetIndex] = null;
+  }
+  }
+  return correctLetterLocations + "-" + correctLetters;
 }
 
+
+
 function mastermind(guess) {
-  // your code here
+  //your code here
+  move++;
+
+  if (guess === solution){
+    return "You guessed it!";
+  }
+var hint = generateHint(solution, guess);
+board.push(hint + " " + guess);
```

Comment:
Booom sauce! Nice work

*Make sure to properly indent lines 68 and 69. They should not be flush with it's mother function.